packages/babel-helper-module-transforms/src/index.js

```diff
@@ -150,6 +155,31 @@ export function buildNamespaceInitStatements(
   return statements;
 }
 
+const getTemplateForReexport = loose => {
+  return loose
+    ? template.statement`EXPORTS.EXPORT_NAME = NAMESPACE.IMPORT_NAME;`
+    : template`
+      Object.defineProperty(EXPORTS, "EXPORT_NAME", {
+        enumerable: true,
+        get: function() {
+          return NAMESPACE.IMPORT_NAME;
+        },
+      });
+    `;
+};
+
+export function buildReexportsFromMeta(meta, metadata, loose) {
+  const templateForCurrentMode = getTemplateForReexport(loose);
```

Comment:
I disagree with @Kovensky here :) I don't think moving this out of the loop matters at all, because there's caching happening here automatically.